src/core/lib/iomgr/port.h

```diff
@@ -38,9 +38,9 @@
 #define GRPC_HAVE_MSG_NOSIGNAL 1
 #define GRPC_HAVE_UNIX_SOCKET 1
 #ifdef LINUX_VERSION_CODE
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 9, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 9, 0)) && defined AF_VSOCK
```

Comment:
I don't think `AF_VSOCK` will be defined anywhere until `socket.h` is imported, we have a chicken and egg problem. Alternatively, we could enable VSOCK for new-enough glibc versions using `#ifdef __GLIBC__` and `#if __GLIBC_PREREQ(2, 18)`.

Reply:
I was to use `__GLIBC_PREREQ` at first, but later I found that the python build will also use musl-libc(`python_musllinux_1_1_x64_cpxx`), so I changed to use `AF_VSOCK` since no method to check musl-libc's version. But just as you commented, this is incorrect here since `socket.h` is not imported.
Updated patch to use `__GLIBC_PREREQ`, please review.

Reply:
I think that's probably fine for now, VSOCK will only be supported when using glibc. If a request comes in to enable this feature for other libc implementations, we can revisit the guards.

Reply:
agree.